src/orchestration/training_cluster_monitor.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2022 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Monitor training clusters.
+
+Restarts training cluster if detects training not progressing.
+"""
+import os
+import re
+import datetime
+import time
+import subprocess
+import sys
+
+from absl import app
+from absl import flags
+from absl import logging
+from google.cloud import storage
+
+FLAGS = flags.FLAGS
+
+
+
+flags.DEFINE_string('project_id', '', 'GCP project id.')
+flags.DEFINE_string('checkpoint_path', '',
+                    'GCS prefix to model checkpoint_file.')
+flags.DEFINE_integer('train_ttl_sec', 2400,
```

Comment:
40 minutes is a bit short time for the large model jobs. Cluster provisioning takes a good portion of that already. Can we increase this to a few hours?

Reply:
Done